pageserver/src/tenant/storage_layer/split_writer.rs

```diff
@@ -296,8 +286,16 @@ impl SplitDeltaLayerWriter {
                     self.generated_layers
                         .push(SplitWriterResult::Discarded(layer_key));
                 } else {
+                    // `finish` will remove the file if anything goes wrong, while we need to handle deleting temporary
+                    // files for `finish_creating`.
                     let (desc, path) = prev_delta_writer.finish(key, ctx).await?;
```

Comment:
I don't understand this comment. We're bailing out here and leaving the already-written finished layer files on disk.

Reply:
the implementation of `finish` will catch the errors and remove the temporary file -> https://github.com/neondatabase/neon/blob/c07cea80bd7a462c108ad04e72fc9af60f31f82b/pageserver/src/tenant/storage_layer/delta_layer.rs#L517-L522